### PR TITLE
Core engine: Abstract the framebuffers

### PR DESCRIPTION
**Pull request by Popov72**
I finally did not use the name `HardwareFramebufferWrapper` to abstract the feature but `RenderTargetWrapper` instead because `framebuffer` is specific to WebGL and does not exist in WebGPU: `RenderTargetWrapper` seems better to me (`RenderTarget` could also be right I think).

Those are breaking changes, but mostly deep in the engine, so most of the users should not be impacted: I only had to update 3 lines of code in the Babylon libraries (2 changes in **inspector/src/textureHelper.ts** and one in **nodeEditor/src/sharedComponents/textureLineComponent.tsx**) and no change in the validation test suite.

As acknowledged, I have moved `updateRenderTargetTextureSampleCount` from `Engine` to `engine.renderTarget.ts`.

I miss to add some docs in the new class + update the **what's new** file, but I'm waiting for your inputs before moving on.

### CODE REVIEW

**Review by deltakosh — APPROVED** [review]

**Comment by RaananW** [comment]
Is this ready to be merged?

**Comment by Popov72** [comment]
It's ok on my side but maybe we should wait for @sebavan review when he comes back from holidays?

**Comment by deltakosh** [comment]
let's wait for seb

File: inspector/src/textureHelper.ts [hunk 1]
```diff
@@ -68,7 +68,7 @@ export class TextureHelper {
         if (rttInternalTexture && internalTexture) {
             const samplingMode = internalTexture.samplingMode;
             texture.updateSamplingMode(Texture.NEAREST_NEAREST_MIPNEAREST);
-            scene.postProcessManager.directRender([lodPostProcess], rttInternalTexture);
+            scene.postProcessManager.directRender([lodPostProcess], rtt.renderTarget);
```
Comment (sebavan) [inline inspector/src/textureHelper.ts:71]: in this case why is the code looking for rttInternalTexture in the previous if ? it is a bit confusing to me ?

**Review by sebavan — COMMENTED** [review]
Would be great to check how bigger is thinengine ???

File: src/Materials/Textures/internalTexture.ts [hunk 2]
```diff
@@ -215,10 +204,6 @@ export class InternalTexture {
     /** @hidden */
     public _compression: Nullable<string> = null;
     /** @hidden */
-    public _generateStencilBuffer: boolean = false;
-    /** @hidden */
-    public _generateDepthBuffer: boolean = false;
-    /** @hidden */
     public _comparisonFunction: number = 0;
```
Comment (sebavan) [inline src/Materials/Textures/internalTexture.ts:207]: Should this still be here as well ?

**Comment by sebavan** [comment]
And the latest point I can think off is the lost context scenario to double check ?

File: src/Materials/Textures/internalTexture.ts (hunk 2 above)
Comment (Popov72) [reply src/Materials/Textures/internalTexture.ts:207]: The `comparisonFunction` is not part of the RTT wrapper(?) We can have a texture which is not a RTT that has a comparison function.

**Comment by Popov72** [comment]
> And the latest point I can think off is the lost context scenario to double check ?

I already did it, I checked 30 or 40 PGs for context lost/restore and they were all ok.

**Review by sebavan — APPROVED** [review]
@RaananW wait on the next commit changing imports to declare :-) from @Popov72

**Comment by Popov72** [comment]
Imports removed!

**Pull request merged**